app/src/main/java/com/example/pix/home/adapters/ChatsAdapter.java

```diff
@@ -93,10 +97,30 @@ public void bind(Chat chat) {
                 this.tvName.setText("" + friend.getUsername());
                 this.tvStatus.setText("" + chat.getStatusText());
                 Message recent = chat.getFirstMessage();
-                this.tvTime.setText("" + (recent != null ? recent.getTime().toString() : ""));
+                this.tvTime.setText("" + (recent != null ? getRelativeTime(recent.getTime().toString()) : ""));
+                tvPix.setText("" + Like.getPix(friend) + "P");
             } catch (ParseException e) {
                 Log.e("Error", "Failed getting status text and/or time", e);
             }
         }
+
+        public String getRelativeTime(String json_response) {
+            //Define the given format
+            String format = "EEE MMM dd HH:mm:ss ZZZZZ yyyy";
+            SimpleDateFormat sf = new SimpleDateFormat(format, Locale.ENGLISH);
+            sf.setLenient(true);
+
+            String relativeDate = "";
+            try {
+                //Get Unix Epoch and get relative from today, then call toString to get readable difference
+                long dateMillis = sf.parse(json_response).getTime();
+                relativeDate = DateUtils.getRelativeTimeSpanString(dateMillis,
+                        System.currentTimeMillis(), DateUtils.SECOND_IN_MILLIS).toString();
+            } catch (java.text.ParseException e) {
+                e.printStackTrace();
```

Comment:
Better error handling. It's probably not necessary to print the stack trace, besides for debugging purposes. In this case, probably just roll back to the raw date.